graph-manager-docs/source/managed-federation/setup.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: Setting up managed federation
+sidebar_title: Setup
+---
+
+import ProjectConfigPanel from 'gatsby-theme-apollo-docs/shared/project-config-panel.mdx';
+
+This article describes how to set up managed federation for an existing data graph that uses [Apollo Federation](https://www.apollographql.com/docs/apollo-server/federation/introduction/).
+
+> As with all changes, you should first set up managed federation in a non-production environment, such as staging. To support this, you can use [variants](../schema-registry/#managing-environments-with-variants), which are distinct versions of the same data graph for different environments.
+
+## 1. Register your implementing services
+
+To enable managed federation, each of your data graph's [implementing services](https://www.apollographql.com/docs/apollo-server/federation/implementing-services/) must register its schema with Graph Manager:
+
+```mermaid
+graph BT;
+  subgraph " "
+  gateway([Gateway]);
+  serviceA[Products service];
+  serviceB[Reviews service];
+  serviceC[Inventory service];
+  end
+  graphManager{{Apollo Graph Manager}};
+  gateway --- serviceA & serviceB & serviceC;
+  serviceA & serviceB & serviceC -. apollo service:push .-> graphManager;
+  class graphManager secondary;
+```
+
+To get set up with schema registration, do the following for each of your implementing services:
+
+1. Complete the following prerequisites:
+
+  <ProjectConfigPanel />
+
+2. In the service's `apollo.config.js` file, add `name` and `url` fields to the `service` object:
+
+    ```json{3-4}:title=apollo.config.js
```

Comment:
This doesn't look right to me. Does any of this work? `service { name: "blah" }` is definitely the **graph-id**, not the implementing service name, which is as it is for legacy reasons, unfortunately :( I have no idea about the `url` or `endpoint` options, but I just have never seen them! Maybe they work all right. Have you tested this guide out?

Reply:
It doesn't work at all! At all!!!

Fixed.